ggshield/secret/secret_scanner.py

```diff
@@ -209,6 +209,9 @@ def _collect_results(
                 )
 
             if not scan.success:
+                if scan.status_code == 403 and scan.detail == "Quota limit reached.":
+                    raise QuotaLimitReachedError()
```

Comment:
We can win a lot of time by doing it here. Here's the different on ggshield repo:
```shell
# Without
time ggshield secret scan repo .
Scanning Commits... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 1243 commits scanned out of 1243 0:00:00

Error: Could not perform the requested action: no more API calls available.
ggshield secret scan repo .  7.31s user 9.87s system 78% cpu 22.028 total

# With
$ time ggshield secret scan repo .
Scanning Commits... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% 1 commits scanned out of 1243 -:--:--

Error: Could not perform the requested action: no more API calls available.
ggshield secret scan repo .  0.26s user 0.11s system 50% cpu 0.731 total
```
Otherwise we have to completely consume `commits_batch` which can take a lot of time.